### ISSUE TITLE
Work with editor code formatters

### ISSUE THREAD

**Issue by paul** [lxxxvi/ruboclean#47]
Most editors can invoke an external command to format a file, but they don't expect the file to change out from under them.

The simplest would be to support a `--stdin` flag, that would read the file in stdin, and output the formatted file to stdout.

FWIW, I was able to get it to work with Neovim & conform.vim using the tempfile feature. I'm not sure how many other editors make that possible, though.

```lua
formatters = {
  ruboclean = {
    condition = function(ctx)
      return vim.fs.basename(ctx.filename) ~= ".rubocop.yml"
    end,
    command = "ruboclean",
    args = { "$FILENAME", "--preserve-comments", "--preserve-paths", "--silent" },
    stdin = false,
  },
},
formatters_by_ft = {
  yaml = { "ruboclean" }
}
```

**Comment by lxxxvi** [comment]
Hi @paul 👋 

Thanks for your input, I appreciate it.

I don't use neovim (or alike), so I may not entirely understand what's going on there. 🙃 

Your idea is to add another way to feed in the YAML data, STDIN instead of reading from a file, and then write the result (the processed YAML) to STDOUT, is that right?

**Comment by paul** [comment]
@lxxxvi Yes, exactly. Similar to how rubocop has a `--stdin` flag: https://docs.rubocop.org/rubocop/usage/basic_usage.html#command-line-flags

**Comment by lxxxvi** [comment]
Hi @paul

Sorry for late answer, I published 0.7.1 few weeks ago, that hopefully implements your requested feature. Would you like to give it a try? Thank you!
